### DISCUSSION TITLE
Updating a file's content without down/upload (i.e. append)?

### DISCUSSION THREAD

**Discussion by Pieparker** [octokit/octokit.js#1961] [Q&A] [answered]
Hiya! Apologies if this isn't the best forum for what I'm about to ask. If that's the case, any signposting would be appreciated.

I've made use of this client to read a file (>10MB) from from my repo, add a line and then commit the updated file.

I'm simply using `octokit.request('https://api.github.com/repos/{owner}/{repo}/git/trees/main')` and `octokit.request('GET /repos/{owner}/{repo}/git/blobs/{file_sha}'` to get the SHA and retrieve the existing commit. I'm then reading the file to a temporary file in an AWS Lambda function and adding the new content. Lastly, I've got `octokit.request('PUT /repos/{owner}/{repo}/contents/{path}'` to commit the new content from the temp file.

My question is... is there any way to append a new line to my existing commit without the down/uploads? Each commit will just be an additional line, guaranteed. I'm not sure if it's possible to declare content as _"existing content plus \n+New content"_ as an option for the PUT request without downloading. TIA!

**Comment by gr2m** [comment, marked as answer]
I don't think that's possible. It might be possible using `git` using the `git apply` command? A good place to ask this question could be https://github.community/c/github-ecosystem/37, there are several GitHub support folks in there. This discussion forum is only about the `@octokit/*` libraries, but your is a good question about the REST API in general

**Reply by Pieparker** [reply]
Totally understandable and I appreciate the signposting!

Truth be told, small world – I started this endeavour using your https://github.com/gr2m/write-csv-file-action however Actions scheduling at the moment is rather unpredictable (i.e. 5 min runs running whenever they please, every 15-60 minutes rather than 5 mins)! I'm down the rabbit hole but I'll use this opportunity to thank you for all your work, both here and there.

**Reply by gr2m** [reply]
Thank you Scott :)
